### DISCUSSION TITLE
BirdNET-Pi not working out of the box in headless setup, local problem or possible bug?

### DISCUSSION THREAD

**Discussion by unosonic** [mcguirepr89/BirdNET-Pi#375] [Q&A] [answered]
I've improvised a mobile setup today and went to a place where there's usually lots of birds activity. But surprise, BNPi does not start recording nor playing a spektrogram. Both services were off and could not turned on via web interface (I had a laptop with me for testing). I've checked the logs later and found, among other errors related to pulseaudio:
`pulseaudio[625987]: Failed to create secure directory (/run/user/1000/pulse): No such file or directory`
So BNPi seems to take the presence of the directory /run/user/1000/ for granted, but after booting and without logging in, it does not necessarily exist. /run/user/$uid is created by pam_systemd and used for storing files used by running processes for a user, eg. pulseaudio. So quick check, I logged in via SSH and everything started working immediately. Is it a problem local to my installation or a bug? 
```
birdy:~/BirdNET-Pi$ env|grep XDG_RUNTIME_DIR
XDG_RUNTIME_DIR=/run/user/1000
```
from the specs:

> $XDG_RUNTIME_DIR defines the base directory relative to which user-specific non-essential runtime files 
> and other file objects (such as sockets, named pipes, ...) should be stored. The directory MUST be owned by 
> the user, and he MUST be the only one having read and write access to it. Its Unix access mode MUST be 0700.
>  
> The lifetime of the directory MUST be bound to the user being logged in. It MUST be created when the user first logs 
> in and if the user fully logs out the directory MUST be removed. If the user logs in more than once he should get pointed 
> to the same directory, and it is mandatory that the directory continues to exist from his first login to his last logout on the 
> system, and not removed in between. Files in the directory MUST not survive reboot or a full logout/login cycle.

**Comment by mcguirepr89** [comment, marked as answer]
Hello, thanks for sharing your observations. I have a few questions and can shed a little light on what you've reported.

The runtime directory does indeed need to be present for PulseAudio to start -- the runtime directory (in this case, the 1000 level regular user) is created when a regular user is logged in.

The installer attempts to configure the system to automatically login to TTY1 via getty by default. This has either failed, or has been disabled.

By any chance, have you either (a) installed the full desktop on top of BirdNET-Pi or perhaps (b) disabled the auto-login to the console on tty0?

Thanks in advance for your help in getting to the bottom of this -- 

My best,
Patrick

**Reply by unosonic** [reply]
Hi Patrick, I've installed Raspberry Pi OS Lite, then BirdNet-Pi with that one-line command. I'havn't disabled auto-login, but I'd assume that it's not enabled by default... Maybe Lite doesn't have an autologin? Is it only because of pulseaudio? Is there need for pulse at all? ok, different topic... 
bests, uno

**Reply by mcguirepr89** [reply]
Autologin is enabled by default for RaspiOS Lite and Full (32 and 64 bit), but I also enable it during installation.

Some commands that might help here:
1. `who`: we're looking for an entry for `tty1` and the shell from which you are issuing that command (probably `pts/0`)
1. `sudo systemctl status getty@tty1.service`: we're hoping for this to be enabled and active (might be a redunant command since we'll know from `who` if it's working or not, but can maybe say why it failed if it is not working)
1. `grep pulse /var/log/syslog | sed "s|$HOSTNAME||" | tail -n30`: To see more log context around `pulse` (and scrub your hostname for the output)

The other thing to test is a regular Full Desktop installation of RaspiOS (which comes with pulse audio installed and in use by default) to test if your mic works outside of BirdNET-Pi (or if you run Debian with either the Debian desktop or Gnome installed, that should also suffice).

**Reply by unosonic** [reply]
Enabling autologin via raspi-config wasn't working for some reason, so I enabled it via `sudo systemctl enable getty@tty1.service`
Now all services are starting after reboot. I'd really prefer not to install a full desktop env ... ;)  
Just in case you're using pulseaudio only in order to access the microphone from more than one client at the same time, that can also be done with alsa only, via dmix plugin.
thx for the hints!

**Comment by ethpi** [comment]
I hope my question fits quite well into this discussion:

Due to security concerns, I had disabled auto-login because the default user has root privileges without a password and is always logged in on tty1. Unfortunately, this causes the recording system to no longer work correctly. Manually creating the directory /run/user/1000 at system startup seems to lead to a working system, but it seems like an unclean solution. 
Is there a proper solution without a logged-in user with root privileges?
